sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/implementation/eventhubs/kafka/KafkaPropertiesBeanPostProcessor.java

```
@@ -43,8 +44,15 @@ public Object postProcessAfterInitialization(Object bean, String beanName) throw
                 + " instead, which can be set as spring.kafka.boostrap-servers=EventHubsNamespacesFQDN:9093.");
 
             KafkaProperties kafkaProperties = (KafkaProperties) bean;
-            String connectionString = connectionStringProvider.getConnectionString();
+            ResolvableType provider = ResolvableType.forClassWithGenerics(ServiceConnectionStringProvider.class, AzureServiceType.EventHubs.class);
+            ObjectProvider<ServiceConnectionStringProvider<AzureServiceType.EventHubs>> beanProvider = applicationContext.getBeanProvider(provider);
+
+            ServiceConnectionStringProvider<AzureServiceType.EventHubs> connectionStringProvider = beanProvider.getIfAvailable();
+            if (connectionStringProvider == null) {
```

should we throw this exception, or should we just return?

---

plus adding some logs?